File: periodoTres/POO/7-03/python/cadastroPessoa.py
# Definição da classe Pessoa
class Pessoa:
    # Método especial de inicialização (__init__)
    # que é chamado ao criar uma nova instância da classe
    def __init__(self, nome, idade, cidade):
        # Inicializa os atributos da instância com os valores passados como argumento
        self.nome = nome
        self.idade = idade
        self.cidade = cidade

    # Método da classe Pessoa formatada com o nome e idade da pessoa
    def apresentar(self):
    #Inprime uma mensagem formatada com o nome e dade da pessoa
        print(f"Olá, meu nome é {self.nome} e tenho {self.idade} anos e moro em {self.cidade}")

# Criando objetos (instâncias) da classe Pessoa
pessoa1 = Pessoa("Alice", 25, "Blumenau")
pessoa2 = Pessoa("Bob", 30, "Curitiba")
pessoa3 = Pessoa("Dudu", 18, "São José dos Pinhais")
pessoa4 = Pessoa("Cássia", 84, "Colombo")

# Chamando o método da classe para cada objeto, exibindo as informações
pessoa1.apresentar()
pessoa2.apresentar()
pessoa3.apresentar()
pessoa4.apresentar()

File: periodoTres/POO/7-03/java/cadastroPessoa/cadastroPessoa.java
// Classe principal que contém o método main (ponto de entrada do programa)
public class cadastroPessoa {
    public static void main(String[] args) {
        // Criando objetos (intâncias) da classe Pessoa
        Pessoa pessoa1 = new Pessoa ("Alice", 25);
        Pessoa pessoa2 = new Pessoa ("Bob", 30);

        // Vhamando método da classe Pessoa para apresentar as informações
        pessoa1.apresentar();
        pessoa2.apresentar();
    }
}

// Definição da classe Pessoa
class Pessoa {
    // Atributos privados da classe Pessoa
    private String nome;
    private int idade;

    // Construtor da classe Pessoa que é chamado ao criar uma nova instância
    public Pessoa(String nome, int idade) {
        // Inicializando os atributos com os valores passados como argumento
        this.nome = nome;
        this.idade = idade;
    }

    // Método da classe Pessoa para apresentar informações
    public void apresentar() {
        // Imprime uma mensagem formatada com o nome e idade da pessoa
        System.out.println("Olá, meu nome é " + nome + " e tenho " + idade + " anos.");
    }

    // Método getter para obter o nome
    public String getNome() {
        return nome;
    }

    // Método setter para definir um novo nome
    public void setNome(String novoNome) {
        nome = novoNome;
    }
}
